File: zemian_scripts/dictLookupSpeed.py
# result:

# With doing nothing (pass)
# >>> 2.1e+02
# 210.0
# >>> _ / 60
# 3.5 mins

# with dict lookup
# >>> 8.05e+02 / 60
# 13.416666666666666 mins

import time, random

# Seepd of empty loop
t1 = time.time()
n = 3_600_000_000
for _ in range(n):
	pass
print(f"{n} iteration - empty loop. Cost {time.time() - t1:.3} secs")

# Speed of dict lookup
data = {}
for i in range(1000):
	data['foo' + str(i)] = i * random.random()

t1 = time.time()
n = 3_600_000_000
count = 0
for _ in range(n):
	if 'foo99' in data:
		count += 1
print(f"{n} iteration - dict lookup of {count}. Cost {time.time() - t1:.3} secs")

File: learn-python-zemian/dictLookupSpeed.java
import java.util.*;

public class dictLookupSpeed {
	public static void main(String[] args) {
		// Speed for empty loop		
		long t1 = System.currentTimeMillis();
		long n = 3_600_000_000L;
		for (long i = 0; i < n; i++) {
		}
		long cost = System.currentTimeMillis() - t1;
		System.out.println(n + " iteration - empty loop. " + 
			"Cost " + (cost / 1000.0) + " secs");

		// Speed of dictionary lookup
		Map<String, Double> data = new HashMap<>();
		for (int i = 0; i < 1000; i++)
			data.put("foo" + i, i * Math.random());
		
		t1 = System.currentTimeMillis();
		long counter = 0;
		for (long i = 0; i < n; i++) {
			if (data.get("foo99") > 0.0) {
				counter ++;
			}
		}
		cost = System.currentTimeMillis() - t1;
		System.out.println(n + " iteration - dict lookup of " + counter + "." +
			" Cost " + (cost / 1000.0) + " secs");
	}
}
